pkg/gktest/printer_go_test.go

```diff
@@ -0,0 +1,257 @@
+package gktest
+
+import (
+	"errors"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestPrinterGo_Print(t *testing.T) {
+	// Not final. While this will eventually exactly match the output of go test,
+	// this is a first pass.
+
+	testCases := []struct {
+		name        string
+		result      []SuiteResult
+		want        string
+		wantVerbose string
+	}{
+		{
+			name:   "no suites",
+			result: []SuiteResult{},
+			want: `PASS
+`,
+			wantVerbose: `PASS
+`,
+		},
+		{
+			name: "empty suite",
+			result: []SuiteResult{{
+				Path: "tests.go",
+			}},
+			want: `ok	tests.go	0s
+PASS
+`,
+			wantVerbose: `ok	tests.go	0s
+PASS
+`,
+		},
+		{
+			name: "empty constraint test",
+			result: []SuiteResult{{
+				Path:    "tests.go",
+				Runtime: Duration(330 * time.Millisecond),
+				TestResults: []TestResult{{
+					Name:    "forbid-labels",
+					Runtime: Duration(330 * time.Millisecond),
+				}},
+			}},
+			want: `ok	tests.go	0.33s
```

Comment:
What goes in to the total runtime? compilation time + evaluation time, or just evaluation time.

IMO separate reporting is most useful, with evaluation time being the most important metric.

Reply:
This will be resolved in later PRs - other ways of printing this information & etc. The point here is to show that we display runtime. As far as printing goes, this exactly matches how "go test" prints runtime (which is the point of this PR).